python/ray/data/tests/test_dataset.py

```diff
@@ -3044,6 +3042,83 @@ def test_groupby_arrow_std(ray_start_regular_shared):
     assert ray.data.from_pandas(pd.DataFrame({"A": [3]})).std("A") == 0
 
 
+def test_groupby_arrow_multicolumn(ray_start_regular_shared):
+    # Test built-in mean aggregation on multiple columns
+    seed = int(time.time())
+    print(f"Seeding RNG for test_groupby_arrow_multicolumn with: {seed}")
+    random.seed(seed)
+    xs = list(range(100))
+    random.shuffle(xs)
```

Comment:
For bonus randomness I'd also `.repartition(random(0, 100))` to induce random partitioning.

Reply:
Otherwise every partition would have 1 element only.

Reply:
I opted to go with a `[1, 10, 100]` partitioning set, I think this should give us the best tradeoff of partitioning edge cases with test time.